### DISCUSSION TITLE
Text-to-Speech

### DISCUSSION THREAD

**Discussion by snxraven** [ggml-org/llama.cpp#461] [Ideas]
I think it would be cool to add a switch to pipe the responses from the models out to esay or another kind of terminal text to speech method. 

This could be very useful to folks with eye sight issues.

**Comment by Sovenok-Hacker** [comment]
I think we can use [Silero TTS](https://github.com/snakers4/silero-models) to do it, but it is written in Python.

**Comment by xor2003** [comment]
[RHVoice](https://github.com/RHVoice/RHVoice)

**Reply by Sovenok-Hacker** [reply]
> [RHVoice](https://github.com/RHVoice/RHVoice)
It is also good, but Silero have better quality.

**Comment by Sovenok-Hacker** [comment]
As I know, Torch works in C++

**Comment by zenixls2** [comment]
mimic3 https://github.com/MycroftAI/mimic3

**Comment by zenixls2** [comment]
[piper](https://github.com/rhasspy/piper) from Mycroft former employee

**Reply by gogo2464** [reply]
Let me just try with

Llamacpp -m model.gguf > out.mp3.

We just have to document tve command. No need to add extra option imho.

**Comment by gianpaj** [comment]
there's some progress tracked here https://github.com/ggml-org/llama.cpp/discussions/13759
